mteb/tasks/Retrieval/eng/DAPFAMPatentRetrieval.py

```diff
@@ -0,0 +1,496 @@
+from __future__ import annotations
+
+from datasets import load_dataset
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+from ....abstasks.TaskMetadata import TaskMetadata
+
+HF_REPO = "datalyes/DAPFAM_patent"
+REFERENCE = "https://arxiv.org/abs/2506.22141"
+BIBTEX = r"""@misc{ayaou2025dapfamdomainawarefamilyleveldataset,
+  archiveprefix = {arXiv},
+  author = {Iliass Ayaou and Denis Cavallucci and Hicham Chibane},
+  eprint = {2506.22141},
+  primaryclass = {cs.CL},
+  title = {DAPFAM: A Domain-Aware Family-level Dataset to benchmark cross domain patent retrieval},
+  url = {https://arxiv.org/abs/2506.22141},
+  year = {2025},
+}"""
+
+_SHARED_METADATA = dict(
+    dataset={"path": HF_REPO, "revision": "780f4011d60297fc6e97a4119b0c516d13afea2d"},
+    reference=REFERENCE,
+    type="Retrieval",
+    modalities=["text"],
+    category="p2p",
+    task_subtypes=["Article retrieval", "Patent retrieval"],
+    eval_splits=["train"],
+    eval_langs=["eng-Latn"],
+    main_score="ndcg_at_100",
+    date=("1964-06-26", "2023-06-20"),
+    domains=["Engineering", "Chemistry", "Legal"],
+    license="cc-by-nc-sa-4.0",
+    annotations_creators="derived",
+    sample_creation="created",
+    dialect=[],
+    bibtex_citation=BIBTEX,
+    is_public=True,
+)
+
+# text-field dictionaries
+_QUERY_FIELDS = {
+    "TitleAbstract": ["title_en", "abstract_en"],
+    "TitleAbstractClaims": ["title_en", "abstract_en", "claims_text"],
+}
+_CORPUS_FIELDS = {
+    "TitleAbstract": ["title_en", "abstract_en"],
+    "TitleAbstractClaims": ["title_en", "abstract_en", "claims_text"],
+    "TitleAbstractClaimsDescription": [
+        "title_en",
+        "abstract_en",
+        "claims_text",
+        "description_en",
+    ],
+}
+
+
+# MIX-IN with shared logic
+class _DAPFAMMixin:
+    # class-level attributes are filled in each concrete subclass
+    domain_filter: str | None = None
+    query_fields: list[str] = []
+    corpus_fields: list[str] = []
+
+    def load_data(self, **kwargs) -> tuple[dict, dict, dict]:
+        ds_c = load_dataset(
+            kwargs.get("dataset", {}).get("path", HF_REPO),
+            "corpus",
+            split=kwargs.get("eval_splits", "train"),
+            revision=kwargs.get("dataset", {}).get(
+                "revision", "780f4011d60297fc6e97a4119b0c516d13afea2d"
+            ),
```

Comment:
Can you change everywhere to this?
```suggestion
            split=self.metadata.eval_splits[0],
            revision=self.metadatadataset["revision"],
```

Reply:
Done, i have also changed the dataset path argument to `self.metadata.dataset["path"]`.